internal/controller/runtime/fsm/runtime_fsm_delete_shoot.go

```diff
@@ -2,14 +2,47 @@ package fsm
 
 import (
 	"context"
+	"encoding/json"
 
 	imv1 "github.com/kyma-project/infrastructure-manager/api/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func sFnDeleteShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl.Result, error) {
+	if !isGardenerCloudDelConfirmationSet(s.shoot.Annotations) {
```

Comment:
The whole section inside this if is a very good candidate for method exstraction. 

if !isGardenerCloudDelConfirmationSet 
   return PatchShootWithAnnotatiuon(s.shoot)

Reply:
As we agreed, we will be changing `Gardener` client in [this PR](https://github.com/kyma-project/infrastructure-manager/pull/275); refactor may be done there.